booster/library/Booster/Pattern/Match.hs

```diff
@@ -227,75 +229,75 @@ match1 _       t1@Injection{}                             t2@KSet{}
 match1 Eval    t1@KSet{}                                  t2@Injection{}                             = addIndeterminate t1 t2
+match1 _       t1@KSet{}                                  t2@Injection{}                             = failWith $ DifferentSymbols t1 t2
 match1 _       t1@KSet{}                                  t2@KMap{}                                  = failWith $ DifferentSymbols t1 t2
 match1 _       t1@KSet{}                                  t2@KList{}                                 = failWith $ DifferentSymbols t1 t2
 match1 _       t1@KSet{}                                  t2@KSet{}                                  = addIndeterminate t1 t2
 match1 _       t1@KSet{}                                  t2@ConsApplication{}                       = failWith $ DifferentSymbols t1 t2
 match1 _       t1@KSet{}                                  t2@FunctionApplication{}                   = addIndeterminate t1 t2
 match1 Rewrite t1@KSet{}                                  (Var t2)                                   = failWith $ SubjectVariableMatch t1 t2
-match1 Eval    t1@KSet{}                                  t2@Var{}                                   = addIndeterminate t1 t2
-match1 Rewrite t1@ConsApplication{}                       (AndTerm t2a t2b)                          = enqueueRegularProblem t1 t2a >> enqueueRegularProblem t1 t2b
+match1 _       t1@KSet{}                                  t2@Var{}                                   = addIndeterminate t1 t2
 match1 Eval    t1@ConsApplication{}                       t2@AndTerm{}                               = addIndeterminate t1 t2
+match1 _       t1@ConsApplication{}                       (AndTerm t2a t2b)                          = enqueueRegularProblem t1 t2a >> enqueueRegularProblem t1 t2b
 match1 _       t1@ConsApplication{}                       t2@DomainValue{}                           = failWith $ DifferentSymbols t1 t2
 match1 _       t1@ConsApplication{}                       t2@Injection{}                             = failWith $ DifferentSymbols t1 t2
 match1 _       t1@ConsApplication{}                       t2@KMap{}                                  = failWith $ DifferentSymbols t1 t2
 match1 _       t1@ConsApplication{}                       t2@KList{}                                 = failWith $ DifferentSymbols t1 t2
 match1 _       t1@ConsApplication{}                       t2@KSet{}                                  = failWith $ DifferentSymbols t1 t2
 match1 matchTy (ConsApplication symbol1 sorts1 args1)     (ConsApplication symbol2 sorts2 args2)     = matchSymbolAplications matchTy symbol1 sorts1 args1 symbol2 sorts2 args2
-match1 Rewrite t1@ConsApplication{}                       t2@FunctionApplication{}                   = addIndeterminate t1 t2
 match1 Eval    (ConsApplication symbol1 sorts1 args1)     (FunctionApplication symbol2 sorts2 args2) = matchSymbolAplications Eval symbol1 sorts1 args1 symbol2 sorts2 args2
```

Comment:
(off-topic) I was stumbling over this line first... why would we try to match those obviously-different things in `Eval` mode? 
After [reading `matchSymbolApplication`](https://github.com/runtimeverification/haskell-backend/pull/3846/files#diff-9255623832855f0b67bd1faa517c9309bc3b13de042a2becd0cbfc17b25bb01bR402) it turns out to be somewhat trivial... since a symbol cannot be both constructor and function, it will always end up in `addIndeterminate`.